src/main/java/org/craftercms/studio/impl/v2/sync/RepositoryWatcherImpl.java

```diff
@@ -147,6 +146,7 @@ private void processRepoEvent(final QueuedEvent queuedEvent) {
             if (!queuedEvent.additionalEvents().get()) {
                 break;
             }
+            queuedEvent.additionalEvents().set(false);
```

Comment:
What's this for?

Reply:
The `additionalEvents` flag is set to true if an event is received while another one is already in the queue for the site.
The idea is to throttle the processing of events when there is constant activity. So the processing will be delayed either until no new events happen during a period (`getEventHandlingDelayMillis()`) or when the max reset count is reached.
We need to clear the flag each time we wait. So processing can proceed if no new event was received in the last sleep period.